File: frontend/src/app/shared/Player/player.ts
import { Card } from "../../module/card/class/card";

export class Player {
    name: string = '';
    hand: Card[] = []; 

    constructor(name: string = '', hand: Card[] = []) {
        this.name = name;
        this.hand = hand;
    }

    public getName(): string {
        return this.name;
    }
    
    public getHand(): Card[] {
        return this.hand;
    }
}

File: backend/src/main/java/com/uno/Uno/Model/Player.java
package com.uno.Uno.Model;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public abstract class Player {
    String sessionId;
    String name;
    List<CardModel> hand;

    public boolean addCard(CardModel card) {
        return this.hand.add(card);
    }
}
